### PR TITLE
test: add `_modifyAllocationsWithChecks`

### PR DESCRIPTION
**Pull request by 0xClandestine**
- TODO: more edge cases (7 failing tests)

### CODE REVIEW

File: src/test/unit/AllocationManagerUnit.t.sol [hunk 1]
```diff
@@ -1501,12 +1591,10 @@ contract AllocationManagerUnitTests_SlashOperator is AllocationManagerUnitTests
         _registerForOperatorSet(defaultOperator, operatorSet);
 
         // Allocate some magnitude, then deallocate some.
-        cheats.startPrank(defaultOperator);
-        allocationManager.modifyAllocations(defaultOperator, allocateParams);
+        _modifyAllocationsWithChecks(defaultOperator, allocateParams);
         cheats.roll(block.number + DEFAULT_OPERATOR_ALLOCATION_DELAY);
-        allocationManager.modifyAllocations(defaultOperator, deallocateParams);
+        _modifyAllocationsWithChecks(defaultOperator, deallocateParams);
```
Comment (ypatil12) [inline src/test/unit/AllocationManagerUnit.t.sol:1596]: Tests should specifically call the write method, not leave it to an internal function

File: src/test/unit/AllocationManagerUnit.t.sol [hunk 2]
```diff
@@ -216,45 +333,32 @@ contract AllocationManagerUnitTests is EigenLayerUnitTestSetup, IAllocationManag
...
-            allocationManager.getAllocations(operator.toArray(), operatorSet, strategy)[0];
-        assertEq(
-            expectedAllocation.currentMagnitude, getAllocations.currentMagnitude, "currentMagnitude != expected"
-        );
-        assertEq(expectedAllocation.pendingDiff, getAllocations.pendingDiff, "pendingDiff != expected");
-        assertEq(expectedAllocation.effectBlock, getAllocations.effectBlock, "effectBlock != expected");
+        // Check `getAllocations` alias for coverge.
+        assertEq(allocationManager.getAllocations(operator.toArray(), operatorSet, strategy)[0], expectedAllocation);
+
+        // TODO: maybe check/track these too?
```
Comment (ypatil12) [inline src/test/unit/AllocationManagerUnit.t.sol:359]: Don't think we need to check/track these

File: src/test/unit/AllocationManagerUnit.t.sol [hunk 3]
```diff
@@ -2372,103 +2190,24 @@ contract AllocationManagerUnitTests_ModifyAllocations is AllocationManagerUnitTe
...
-        );
-
         // Create and register for a new operator set with the same default strategy.
         // If we try to allocate to this new set, it should clear the deallocation queue,
         // allowing all magnitude to be allocated
         OperatorSet memory finalOpSet = _createOperatorSet(OperatorSet(defaultAVS, r.Uint32()), defaultStrategies);
         _registerForOperatorSet(defaultOperator, finalOpSet);
-        AllocateParams[] memory finalAllocParams = _newAllocateParams(finalOpSet, WAD);
-
-        _checkClearDeallocationQueueEvents({
```
Comment (ypatil12) [inline src/test/unit/AllocationManagerUnit.t.sol:2441]: Commenting this out is causing failures likely since deallocation events are different

**Review by ypatil12 — CHANGES_REQUESTED** [review]
Minor changes, looking clean overall

File: src/test/unit/AllocationManagerUnit.t.sol (hunk 1 above)
Comment (ypatil12) [reply src/test/unit/AllocationManagerUnit.t.sol:1596]: Let's call `modifyAllpocations` in the actual test and then leave checks to an internal function
